src/pmstorage.cc

```diff
@@ -0,0 +1,254 @@
+  try {
+    inventory.Push(mat);
+  } catch (cyclus::Error& e) {
+    e.msg(Agent::InformErrorMsg(e.msg()));
+    throw e;
+  }
+
+
+  LOG(cyclus::LEV_INFO5, "ComCnv")
+      << prototype() << " added " << mat->quantity()
+      << " of packagedmatl to its inventory, which is holding "
+      << inventory.quantity() << " total.";
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void PmStorage::BeginProcessing_() {
+
+  while (inventory.count() > 0) {
+    try {
+      processing.Push(inventory.Pop());
+      entry_times.push_back(context()->time());
+
+      LOG(cyclus::LEV_DEBUG2, "ComCnv")
+          << "PmStorage " << prototype()
+          << " added resources to processing at t= " << context()->time();
+    } catch (cyclus::Error& e) {
+      e.msg(Agent::InformErrorMsg(e.msg()));
+      throw e;
+    }
+
+
+  }
+
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void PmStorage::ProcessMat_(double cap) {
+  using cyclus::PackagedMaterial;
+  using cyclus::ResCast;
+  using cyclus::toolkit::ResBuf;
+  using cyclus::toolkit::Manifest;
+
+  if (!ready.empty()) {
+    try {
+      double max_pop = std::min(cap, ready.quantity());
+
+      if (discrete_handling) {
+        if (max_pop == ready.quantity()) {
+          stocks.Push(ready.PopN(ready.count()));
+        } else {
+          double cap_pop = ready.Peek()->quantity();
+          while (cap_pop <= max_pop && !ready.empty()) {
+            stocks.Push(ready.Pop());
+            cap_pop += ready.empty() ? 0 : ready.Peek()->quantity();
+          }
+        }
+      } else {
+        stocks.Push(ready.Pop(max_pop, cyclus::eps_rsrc()));
+      }
+
+
+      LOG(cyclus::LEV_INFO1, "ComCnv") << "PmStorage " << prototype()
+                                       << " moved resources"
+                                       << " from ready to stocks"
+                                       << " at t= " << context()->time();
+    } catch (cyclus::Error& e) {
+      e.msg(Agent::InformErrorMsg(e.msg()));
+      throw e;
+    }
+  }
+}
+
+//- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+void PmStorage::ReadyMatl_(int time) {
+  using cyclus::toolkit::ResBuf;
+
+  int to_ready = 0;
+
+  while (!entry_times.empty() && entry_times.front() <= time) {
+    entry_times.pop_front();
+    ++to_ready;
+  }
+
+  ready.Push(processing.PopN(to_ready));
+
+}
+
+void PmStorage::RecordPosition() {
+  std::string specification = this->spec();
+  context()
+      ->NewDatum("AgentPosition")
+      ->AddVal("Spec", specification)
+      ->AddVal("Prototype", this->prototype())
+      ->AddVal("AgentId", id())
+      ->AddVal("Latitude", latitude)
+      ->AddVal("Longitude", longitude)
+      ->Record();
+}
+
+
+// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+extern "C" cyclus::Agent* ConstructPmStorage(cyclus::Context* ctx) {
+  return new PmStorage(ctx);
+}
+
+} // namespace pmstorage
```

Comment:
Should be a blank line here, missing commit?
```suggestion
} // namespace pmstorage

```